api/src/services/lottery.service.ts

```diff
@@ -956,6 +974,64 @@ export class LotteryService {
     return formattedActivityLogs;
   }
 
+  /**
+    runs the job to auto expire lotteries that are passed their due date
+    will call the the cache purge to purge all listings as long as updates had to be made
+  */
+  async autoPublishResults(): Promise<SuccessDTO> {
+    this.logger.warn('autoPublishLotteryResults job running');
+    await this.listingService.markCronJobAsStarted(
+      LOTTERY_PUBLISH_CRON_JOB_NAME,
+    );
+    const tomorrow = dayjs(new Date()).add(1, 'days').toDate();
+    const releasedListings = await this.prisma.listings.findMany({
+      select: {
+        id: true,
+        name: true,
+        jurisdictions: true,
+      },
+      where: {
+        status: ListingsStatusEnum.closed,
+        reviewOrderType: ReviewOrderTypeEnum.lottery,
+        lotteryOptIn: true,
+        lotteryStatus: LotteryStatusEnum.releasedToPartners,
+        listingEvents: {
+          some: {
+            type: ListingEventsTypeEnum.publicLottery,
+            startDate: { lt: tomorrow },
```

Comment:
In your description it mentions that it should be at end of day. We don't need to take the start or end time in to consideration?